### PR TITLE
disable zstd support added

### PR DESCRIPTION
**Pull request by sulincix**
This changes allow to disable zstd support.

Default status enabled.

### CODE REVIEW

**Comment by hughsie** [comment]
Ohh wow, you've got an environment without zstd?

**Comment by sulincix** [comment]
Yes. I am not using zstd :D

File: src/meson.build [hunk 1]
```diff
@@ -39,6 +39,11 @@ subdir('libxmlb')
 
 mapfile = 'libxmlb.map'
 vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
+if get_option('zstd')
+    zstd_sources = ['xb-zstd-decompressor.c']
+else
+    zstd_sources = []
```
Comment (hughsie) [inline src/meson.build:45]: maybe call this extra_sources or something? I could imagine other things there too.

Comment (sulincix) [reply src/meson.build:45]: Okay. I am changing now.

Comment (hughsie) [reply src/meson.build]: also, maybe define the [] before the `if get_op` and then use += for the xb-zstd-decompressor

Comment (sulincix) [reply src/meson.build]: https://github.com/hughsie/libxmlb/pull/134/commits/9a751e1576b8fa68d8fe3b41a8d81d0341fd6e3f fixed

Comment (sulincix) [reply src/meson.build]: I think if we will add extra features. we can use extra_sources array. 
Also extra_sources used 2 times in file...

File: meson.build [hunk 2]
```diff
@@ -119,7 +119,6 @@ mandir = join_paths(prefix, get_option('mandir'))
 gio = dependency('gio-2.0', version : '>= 2.45.8')
 giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
 lzma = dependency('liblzma')
-zstd = dependency('libzstd')
```
Comment (ximion) [inline meson.build:122]: You could simplify the code by just using something like this:
``` meson
zstd = dependency('libzstd')
if get_option('zstd')
    add_project_arguments('-DENABLE_ZSTD', language: 'c')
    if not zstd.found() 
        error('libzstd is missing, please install it to continue!')
    endif
endif
``` 
That avoids having the extra `libxmlb_deps` variable.
By adding a `HAVE_ZSTD` define directly to the project's `configuration_data` above, you can simplify the code even more and have an even smaller Meson definition (with only `extra_sources remaining).

Comment (hughsie) [reply meson.build:122]: I think you could use a meson `feature` too.

Comment (sulincix) [reply meson.build:122]: https://github.com/hughsie/libxmlb/pull/134/commits/f72a109986f0d2ad6dedf4239a7489272bd24b21

File: meson.build [hunk 3]
```diff
@@ -120,6 +120,12 @@ gio = dependency('gio-2.0', version : '>= 2.45.8')
 giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
 lzma = dependency('liblzma')
 zstd = dependency('libzstd')
+if get_option('zstd')
+    if not zstd.found() 
+        error('libzstd is missing, please install it to continue!')
+    endif
+      conf.set('HAVE_ZSTD', 1)
```
Comment (hughsie) [inline meson.build:127]: i think the indenting is wonky here -- two less spaces

Comment (sulincix) [reply meson.build:127]: https://github.com/hughsie/libxmlb/blob/main/meson.build#L144 I wrote like this

Comment (sulincix) [reply meson.build]: https://github.com/hughsie/libxmlb/pull/134/commits/a074f2e283c2333a317f3fbb7fad4d1677fea4da fixed.

File: meson.build [hunk 4]
```diff
@@ -120,6 +120,12 @@ gio = dependency('gio-2.0', version : '>= 2.45.8')
 giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
 lzma = dependency('liblzma')
 zstd = dependency('libzstd')
+if get_option('zstd')
+    if not zstd.found() 
+        error('libzstd is missing, please install it to continue!')
```
Comment (eli-schwartz) [inline meson.build:125]: But zstd is mandatory by default... this is why a feature option was suggested.

**Pull request merged**